src/app/clusters/wifi-network-diagnostics-server/wifi-network-diagnostics-server.cpp

```diff
@@ -57,19 +57,20 @@ class WiFiDiagosticsAttrAccess : public AttributeAccessInterface
     CHIP_ERROR ReadWiFiRssi(AttributeValueEncoder & aEncoder);
 };
 
-template <typename T>
-CHIP_ERROR WiFiDiagosticsAttrAccess::ReadIfSupported(CHIP_ERROR (DiagnosticDataProvider::*getter)(T &),
+template <typename T, typename Type>
+CHIP_ERROR WiFiDiagosticsAttrAccess::ReadIfSupported(CHIP_ERROR (DiagnosticDataProvider::*getter)(T &), Type & data,
                                                      AttributeValueEncoder & aEncoder)
 {
-    T data;
-    CHIP_ERROR err = (DeviceLayer::GetDiagnosticDataProvider().*getter)(data);
-    if (err == CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE)
+    T value;
+    CHIP_ERROR err = (DeviceLayer::GetDiagnosticDataProvider().*getter)(value);
+
+    if (err == CHIP_NO_ERROR)
     {
-        data = 0;
+        data.SetNonNull(value);
```

Comment:
But this assumes the type is Nullable, no?  Why not just have `Nullable<T>` declared inside this function instead of the extra thing?  Is it just so we make sure that the type matches the actual attribute type?  But how do we know that `T` is the right thing for the type then?

Reply:
Yes, all attributes read via ReadIfSupported are Null-able.

```
    template <typename T, typename Type>
    CHIP_ERROR ReadIfSupported(CHIP_ERROR (DiagnosticDataProvider::*getter)(T &), Type & data, AttributeValueEncoder & aEncoder);
```

The typename T and typename Type need to match, and this binding is done at the calling place.

```
 Attributes::BeaconLostCount::TypeInfo::Type count;
 return ReadIfSupported(&DiagnosticDataProvider::GetWiFiBeaconLostCount, count, aEncoder);
```
 
We need to encode the data with defined type to be correct, right? It Nullable<T> and  Attributes::BeaconLostCount::TypeInfo::Type<T> equivalent?